simple-wallet/src/lib.rs

```diff
@@ -154,8 +155,14 @@ where
         let fee = (weight * fee_rate) / 1000;
         tx.output[0].value -= fee;
 
+        // construct psbt
+        let mut psbt = PartiallySignedTransaction::from_unsigned_tx(tx.clone()).unwrap();
         for (i, utxo) in utxos.iter().enumerate().take(tx.input.len()) {
-            self.sign_tx_input(&mut tx, i, &utxo.tx_out, None)?;
+            psbt.inputs[i].witness_utxo = Some(utxo.tx_out.clone());
+        }
+
+        for (i, _) in utxos.iter().enumerate().take(tx.input.len()) {
+            self.sign_psbt_input(&mut psbt, i)?;
```

Comment:
I think you're never feeling the `tx` witness? (wow using PSBT is so error prone ;) ) Probably nice to use `extract_tx` here as well?

Reply:
Good catch haha, needed to use the PSBT more :joy: